### PR TITLE
Add Stick / Rename Vice Grip

### PR DESCRIPTION
**Pull request by PartMan7**
NOTE: Unsure if this is desired, or if Stick is preferably going to be retained as Stick while adding Leek as an alias for it.

(Should ``user.baseTemplate.species === 'Farfetch\'d' || user.baseTemplate.species == 'Farfetch\'d-Galar'`` be replaced with ``user.baseTemplate.baseSpecies === 'Farfetch\'d'``?)

### CODE REVIEW

**Comment by HoeenCoder** [comment]
> (Should `user.baseTemplate.species === 'Farfetch\'d' || user.baseTemplate.species == 'Farfetch\'d-Galar'` be replaced with `user.baseTemplate.baseSpecies === 'Farfetch\'d'`?)

No, baseSpecies is for formes.
```
>> Dex.getTemplate('sirfetchd').baseSpecies
<< "Sirfetch'd"
```

File: data/items.js [hunk 1]
```diff
@@ -3117,6 +3117,22 @@ let BattleItems = {
...
 	},
+	"leek": {
+		id: "leek",
+		name: "Leek",
+		fling: {
+			basePower: 60,
+		},
+		spritenum: 475,
+		onModifyCritRatio(critRatio, user) {
+			if (user.baseTemplate.species === 'Farfetch\'d' || user.baseTemplate.species == 'Farfetch\'d-Galar' || user.baseTemplate.species == 'Sirfetch\'d') {
```
Comment (HoeenCoder) [inline data/items.js]:
```suggestion
			if (user.baseTemplate.species === 'Farfetch\'d' || user.baseTemplate.species === 'Farfetch\'d-Galar' || user.baseTemplate.species === 'Sirfetch\'d') {
```
Always use strict equals.

**Comment by PartMan7** [comment]
I meant baseSpecies for Farfetch'd and Farfetch'd-Galar.

**Comment by Zarel** [comment]
Historically, we've just renamed the item/move in question and added an alias, but Stick -> Leek is definitely the biggest rename we've ever seen in the history of renames. We didn't rename PRZCureBerry to Cheri Berry, after all.

So I guess I'm fine with this implementation.

Do you want to handle the Vice Grip -> Vise Grip rename here, too? That one is just a rename and an alias from the old name.

**Comment by PartMan7** [comment]
Alright, on it.

**Comment by PartMan7** [comment]
I think I missed the second ===, will fix when I get back to the PC.

File: data/items.js [hunk 2]
```diff
@@ -3117,6 +3117,22 @@ let BattleItems = {
 		desc: "Evolves Gloom into Vileplume, Weepinbell into Victreebel, Exeggcute into Exeggutor or Alolan Exeggutor, Eevee into Leafeon, Nuzleaf into Shiftry, and Pansage into Simisage when used.",
 		shortDesc: "Evolves certain species of Pokemon when used.",
 	},
+	"leek": {
+		id: "leek",
+		name: "Leek",
+		fling: {
+			basePower: 60,
+		},
+		spritenum: 475, //TODO: Sprite, using Stick as a placeholder
```
Comment (Zarel) [inline data/items.js]:
```suggestion
		spritenum: 475,
```

No placeholder, it's literally a Stick.

File: data/items.js [hunk 3]
```diff
@@ -3117,6 +3117,22 @@ let BattleItems = {
...
+		onModifyCritRatio(critRatio, user) {
+			if (user.baseTemplate.species === 'Farfetch\'d' || user.baseTemplate.species === 'Farfetch\'d-Galar' || user.baseTemplate.species === 'Sirfetch\'d') {
```
Comment (Zarel) [inline data/items.js]:
```suggestion
			if (["Farfetch'd", "Sirfetch'd"].includes(user.baseTemplate.baseSpecies)) {
```

Comment (Zarel) [reply data/items.js]: Also are we sure this is `baseTemplate` and not `template`?

File: data/items.js (hunk 2 above)
Comment (PartMan7) [reply data/items.js]: Wasn't sure.

HoeenHero - Last Friday at 12:25 AM
Not sure if we have a sprite id for leek yet but is is different

File: data/items.js (hunk 3 above)
Comment (PartMan7) [reply data/items.js]: All other implementations of species-specific items (Light Ball, Soul Dew) use ``baseTemplate``

**Comment by PartMan7** [comment]
Notes: 

Placeholder: HoeenHero - Last Friday at 12:25 AM
Not sure if we have a sprite id for leek yet but is is different

baseTemplate: All other implementations (Light Ball, Soul Dew, etc) use ``baseTemplate``, so I stuck with that.

File: data/items.js (hunk 2 above)
Comment (Zarel) [reply data/items.js]: I might be misinterpreting HoeenHero, but he's wrong about that.

**Pull request merged**